src/main/java/com/study/cheeper/model/dto/UserDto.java

```diff
@@ -0,0 +1,30 @@
+package com.study.cheeper.model.dto;
+
+import com.study.cheeper.model.User;
+
+public class UserDto {
+
+    private Integer id;
+    private String name;
+    private String bio;
+    private String image;
+    private int numberOfCheeps;
+
+    public UserDto(User user) {
+        this.id = user.getId();
+        this.name = user.getName();
+        this.bio = user.getBio();
+        this.image = user.getImage();
+        this.numberOfCheeps = user.getNumberOfCheeps();
+    }
+
+    public Integer getId() { return id; }
+
+    public String getName() { return name; }
+
+    public String getBio() { return bio; }
+
+    public String getImage() { return image; }
+
+    public int getNumberOfCheeps() { return numberOfCheeps; }
+}
```

Comment:
Aqui tem dois pontos... Primeiro que não faça mais sentido manter User e UserDTO e sim Profile e ProfileDTO.

E para essa parte funcionar tive que deixar a lista de cheeps como EAGER no User... ainda to vendo como resolver isso e as outras coisas

Reply:
Vi que você adicionou o `count` com um parâmetro separado lá no `ProfileController`.  Pra trazer esse count na entidade  você pode usar uma annotations `@Formula` do hibernate, ela te permite preencher campos das entidades com algum código SQL. : https://vladmihalcea.com/how-to-map-calculated-properties-with-jpa-and-hibernate-formula-annotation/

Outra alternativa seria não usar as entidades para leitura. Como as páginas raramente são um para um com as entidades, pode ser interessante fazer queries específicas para popular determinadas páginas ou relatórios, ao invés de trazer a entidade. A idéia é fazer um DTO que representa por exemplo a página e preencher diretamente do banco de dados: https://vladmihalcea.com/dto-projection-jpa-query/

Isso é interessante pois te dá bastante controle sobre o que é buscado do banco, e evita os problemas de lazyload e n+1, etc.